modules/health-check/src/main/scala/higherkindness/mu/rpc/healthcheck/HealthService.scala

```diff
@@ -0,0 +1,97 @@
+package higherkindness.mu.rpc.healthcheck
+
+import cats._
+import cats.effect._
+import cats.syntax.all._
+import _root_.fs2._
+import _root_.fs2.concurrent._
+import _root_.grpc.health.v1.health._
+import _root_.grpc.health.v1.health.HealthCheckResponse.ServingStatus
+import io.grpc.{Status, StatusException}
+
+trait HealthService[F[_]] extends Health[F] {
+
+  def setStatus(serviceStatus: ServiceStatus): F[Unit]
+
+  def clearStatus(service: String): F[Unit]
+
+}
+
+object HealthService {
+
+  def build[F[_]: Concurrent]: F[HealthService[F]] = {
+    val checkRef: F[Ref[F, Map[String, ServingStatus]]] =
+      // empty string indicates the health of the server in general,
+      // rather than any particular gRPC service running on the server
+      Ref.of[F, Map[String, ServingStatus]](Map("" -> ServingStatus.SERVING))
+
+    val watchSignal: F[SignallingRef[F, ServiceStatus]] =
+      SignallingRef.of[F, ServiceStatus](
+        ServiceStatus("DummyService", HealthCheckResponse.ServingStatus.UNKNOWN)
```

Comment:
Where this comes from (the `DummyService` name)?

Reply:
This is a bit of a hack, just because we have to initialize the `SignallingRef` with *something*.

I suppose the alternative would be to make it a `SignallingRef[F, Option[ServiceStatus]]`, initialise it with `None`, and wrap all service status updates in `Some(...)`. I'll try that and see how it looks.

Reply:
Yep, that's a bit nicer.